File: lib/admin/profile-builder.test.ts
/** Unit tests for admin profile builder. */

import assert from "node:assert/strict";

import { buildDefaultProfile } from "@/lib/admin/profile-builder";

const agenceProfile = buildDefaultProfile(
  { besoin: "Site web", droit_retractation: true },
  "agence",
);

assert.equal((agenceProfile.form as { besoin: string }).besoin, "Site web");
assert.equal(
  (agenceProfile.communication as { delays: { retraction_days: number } }).delays
    .retraction_days,
  4,
);
assert.equal((agenceProfile.match as { active_rdv: boolean }).active_rdv, false);
assert.equal(
  (agenceProfile.display as { timeline: unknown[] }).timeline.length,
  4,
);

const entrepriseProfile = buildDefaultProfile({ besoin: "Refonte" }, "entreprise");
assert.equal(
  (entrepriseProfile.communication as { delays: { retraction_days: number } }).delays
    .retraction_days,
  0,
);
assert.equal(
  (entrepriseProfile.display as { timeline: { label: string }[] }).timeline[0].label,
  "Qualification de votre besoin",
);

console.log("profile-builder.test.ts: ok");

File: app/streamlit_funnels/fiches/profile_builder.test.py
"""Unit tests for profile_builder."""

from __future__ import annotations

import sys
from pathlib import Path

_APP_DIR = Path(__file__).resolve().parent.parent
if str(_APP_DIR) not in sys.path:
    sys.path.insert(0, str(_APP_DIR))

from fiches.profile_builder import build_default_profile


def test_agence_profile_shape() -> None:
    profile = build_default_profile(
        {"besoin": "Site web", "droit_retractation": True},
        "agence",
    )
    assert profile["form"]["besoin"] == "Site web"
    assert profile["communication"]["delays"]["retraction_days"] == 4
    assert profile["match"]["active_rdv"] is False
    assert len(profile["display"]["timeline"]) == 4


def test_entreprise_profile_shape() -> None:
    profile = build_default_profile({"besoin": "Refonte"}, "entreprise")
    assert profile["communication"]["delays"]["retraction_days"] == 0
    assert profile["display"]["timeline"][0]["label"] == "Qualification de votre besoin"


if __name__ == "__main__":
    test_agence_profile_shape()
    test_entreprise_profile_shape()
    print("profile_builder.test.py: ok")
